components/ArticleList/ArticleList.js

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import useSWR from "swr";
+import ArticleCard from "../ArticleCard/ArticleCard";
+
+export default function ArticleList() {
+  const { data, error } = useSWR("https://www.tagesschau.de/api2/news");
```

Comment:
It is good practice to keep the components as dump as can be. So please make the fetch in pages/index.js and send the data as a prop to the component.

Reply:
I've made some changes based on your suggestions. The data fetching now resides in the index.js and the favorite button logic has been moved to the ArticleCard component. I believe this structure provides a clearer separation of concerns and allows each component to handle its specific responsibility, while keeping the FavoriteButton as dumb as possible.